### PR TITLE
Indexer development

### LINKED ISSUE #7: Indexer Subsystem
Build Indexer Subsystem

### PR DESCRIPTION
**Pull request by adarsh-cms**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/IndexSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    private void setState(IndexerState indexerState) {
+        switch (indexerState) {
+            case FEED -> indexMotor.set(FeederConstants.FEEDSPEED);
+            case INDEX -> indexMotor.set(FeederConstants.INDEXSPEED);
+            case EJECT -> indexMotor.set(FeederConstants.EJECTSPEED);
+            case SOFTFEED -> indexMotor.set(FeederConstants.SOFTFEEDSPEED);
+            case BREAK -> indexMotor.set(FeederConstants.BREAKSPEED);
+        }
+        currentState = indexerState;
+    }
```
Comment (drewbxyz) [inline src/main/java/frc/robot/subsystems/IndexSubsystem.java]: Why do we need both functions here? Both accomplish similar things.

File: src/main/java/frc/robot/subsystems/IndexSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.hardware.TalonFX;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class IndexSubsystem extends SubsystemBase {
+    private final static IndexSubsystem INSTANCE = null;
```
Comment (drew-beamer) [inline src/main/java/frc/robot/subsystems/IndexSubsystem.java:11]: This is never used

**Review by drewbxyz — COMMENTED** [review]
See comment

File: src/main/java/frc/robot/subsystems/IndexSubsystem.java (hunk 1 above)
Comment (samperlmutter) [inline src/main/java/frc/robot/subsystems/IndexSubsystem.java]: This method doesn't actually set the state, it gets the state then sets the `indexMotor` speed accordingly. These can also be consolidated into methods on the enum

File: src/main/java/frc/robot/subsystems/IndexSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        SmartDashboard.putBoolean("indexer/Left sensor enabled", true);
+        SmartDashboard.putBoolean("indexer/Right sensor enabled", true);
+
+    }
+
+    public boolean haveNote() {
+        return (SmartDashboard.getBoolean("indexer/Left sensor enabled", true) && !leftIndexSensor.get()) ||
+        (SmartDashboard.getBoolean("indexer/Right sensor enabled", true) && !rightIndexSensor.get());
+
+    }
```
Comment (samperlmutter) [inline src/main/java/frc/robot/subsystems/IndexSubsystem.java]: We shouldn't be basing whether we have the note on the value in smart dashboard. smart dashboard should only represent the state after the fact.

**Review by samperlmutter — CHANGES_REQUESTED** [review]

**Review by samperlmutter — APPROVED** [review]

**Pull request merged**